rdp-proto/src/tpdu.rs

```diff
@@ -23,6 +24,39 @@ pub enum X224TPDUType {
     Error = 0x70,
 }
 
+pub fn decode_x224(input: &mut BytesMut) -> io::Result<(X224TPDUType, BytesMut)> {
+    let mut stream = input.as_ref();
+    let len = read_tpkt_len(&mut stream)? as usize;
+
+    if input.len() < len {
+        return Err(io::Error::new(
+            io::ErrorKind::UnexpectedEof,
+            "The buffer length is less then the real length",
+        ));
+    }
+
+    let (_, code) = parse_tdpu_header(&mut stream)?;
+
+    let mut tpdu = input.split_to(len as usize);
+    match code {
+        X224TPDUType::Data => tpdu.advance(TPDU_DATA_LENGTH),
```

Comment:
`advance` panics if there is not enough data remaining. I didn't see a safe advance function in the library, I think we need to manually check the remaining length before calling advance.

https://docs.rs/bytes/0.4.12/bytes/struct.BytesMut.html#method.advance

Reply:
Good catch, fixed it